### PR TITLE
feat: improve ux of debug cli

### LINKED ISSUE #725: Improve debug subcommand interface
currently, we prefix all the debug CLI commands with `!` in order to follow the same pattern of the REPL interface

https://github.com/0xPolygonMiden/miden-vm/blob/985a7aff51fb98cebe3cd82b4371d14f9a3b8376/miden/src/cli/debug/command.rs#L24-L44

However, this is not ergonomics as it makes the debug experience much harder to interact with. CLI debuggers often follows a similar interface to `gdb`, and will allow the user to interact with his program using a minimalist input.

I suggest the commands are updated to

```rust
"n" | "next" => Ok(Self::Play(1)),
"c" | "continue" => Ok(Self::PlayAll),
"b" | "back" => Ok(Self::Rewind(1)),
"q" | "quit" => Ok(Self::Quit),
"h" | "help" => Ok(Self::Help),
"r" | "rewind" => Ok(Self::RewindAll),
"p" | "print" => Ok(Self::PrintState),
"m" | "mem" => Ok(Self::PrintMem),
"s" | "stack" => Ok(Self::PrintStack),
"l" | "clock" => Ok(Self::Clock),
```

We should probably have a completer providing hints for rustyline as well. This can be done via [rustyline::Hinter](https://docs.rs/rustyline/11.0.0/rustyline/hint/trait.Hinter.html)

### PR DESCRIPTION
**Pull request by vlopes11**
closes #725

### CODE REVIEW

File: miden/src/cli/debug/mod.rs [hunk 1]
```diff
@@ -56,18 +56,28 @@ impl DebugCmd {
...
                             break;
                         }
                     }
-                    Err(err) => println!("{err}"),
+                    Ok(None) => (),
+                    Err(err) => eprintln!("{err}"),
                 },
-                Err(err) => println!("malformed command - failed to read user input: {}", err),
+                Err(ReadlineError::Interrupted) => {
+                    //eprintln!("CTRL-C");
```
Comment (frisitano) [inline miden/src/cli/debug/mod.rs:74]: What is the expected behaviour here? Did you intend on adding a print or a break?

**Review by frisitano — COMMENTED** [review]
Thank you for this, it makes a huge difference to the ergonomics!  Whizzing around the debugger has never felt so fluid.  I left a question inline regarding behaviour for interrupts.

File: miden/src/cli/debug/executor.rs [hunk 2]
```diff
@@ -180,23 +180,36 @@ impl DebugExecutor {
...
-            !stack       displays the complete state of the stack\n\
-            !stack[i]    displays the stack element at index i\n\
-            !mem         displays the complete state of memory\n\
-            !mem[i]      displays memory at address i\n\
-            !clock       displays the current clock cycle\n\
-            !quit        quits the debugger\n\
-            !help        displays this message";
+            ---------------------------------------------------------------------\n\
+            next               steps to the next clock cycle\n\
+            next <c>           steps to the next `c` clock cycles\n\
```
Comment (frisitano) [inline miden/src/cli/debug/executor.rs:187]: I'm not sure if `steps to the next `c` clock cycles` reads well.  Maybe we could change it to:
```
            next                  progress `1` clock cycle\n\
            next <c>           progress `c` clock cycles\n\
```

What do you think?

File: miden/src/cli/debug/executor.rs [hunk 3]
```diff
@@ -180,23 +180,36 @@ impl DebugExecutor {
...
+            print              displays the complete state of the virtual machine\n\
+            print mem          displays the complete state of memory\n\
+            print mem <i>      displays memory at address `i`\n\
+            print stack        displays the complete state of the stack\n\
+            print stack <i>    displays the stack element at index `i`\n\
+            clock              displays the current clock cycle\n\
+            quit               quits the debugger\n\
+            help               displays this message\n\
+            \n\
+            The following mappings are also available:\n\
```
Comment (frisitano) [inline miden/src/cli/debug/executor.rs:201]: I wonder if we could format the short form commands in line.  Maybe we create an extra column for the short form and add titles to the columns something like:
```
Long                      |      Short      |                  Description
----------------------------------------------------------------------------------------------
print mem <i>                     p m <i>             displays memory at address `i`\n\
```

**Review by frisitano — COMMENTED** [review]
Added a few more comments.

File: miden/src/cli/debug/executor.rs (hunk 2 above)
Comment (bobbinth) [reply miden/src/cli/debug/executor.rs:187]: I'd probably do something like:

```
next          moves to the next clock cycle\n\
next <n>      moves `n` clock cycles forward\n\
```

**Review by bobbinth — APPROVED** [review]
Looks great! Thank you!

File: miden/src/cli/debug/mod.rs (hunk 1 above)
Comment (vlopes11) [reply miden/src/cli/debug/mod.rs:74]: Better replace this by a comment. The idea is to do nothing when the user inputs ctrl+c, and halt whatever he is inputting.

File: miden/src/cli/debug/executor.rs (hunk 2 above)
Comment (vlopes11) [reply miden/src/cli/debug/executor.rs:187]: Done!

File: miden/src/cli/debug/executor.rs (hunk 3 above)
Comment (vlopes11) [reply miden/src/cli/debug/executor.rs:201]: Printing tables on CLI is not that trivial as it depends on the font of choice, the characteristics of the terminal, etc.

We do have some crates out there that will take care of all of that for us, but I wonder if it's worthy to add another dependency. Wdyt if we leave this as open issue for future improvement?

If we decide to include in this PR, I think the best call is https://crates.io/crates/prettytable-rs

Comment (frisitano) [reply miden/src/cli/debug/executor.rs:201]: yeah I think this is more of a nice to have then a hard requirement.  It doesn't really affect UX, it just makes it a little more pretty but that not hugely important.  I don't think it's worth spending time on now if it's not simple.

**Review by frisitano — APPROVED** [review]
This change makes a huge difference. Thanks!

**Pull request merged**